### PR TITLE
Turn exercise 003 into practice

### PR DESCRIPTION
**Pull request by mnaibei**
This PR is in reference to task #2167

I have closed the previous PR and opened a new one @cuihtlauac 

Fix Issue #2167

### CODE REVIEW

File: .DS_Store
Comment (cuihtlauac) [inline .DS_Store:1]: This file should not be part of this PR, please remove it.

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Thanks @mnaibei. `dune build --root .` does not pass. Please fix

**Comment by cuihtlauac** [comment]
@mnaibei are you still working on this?

**Comment by mnaibei** [comment]
@cuihtlauac yes I am., sorry for the delay I had some technical difficulties

**Comment by mnaibei** [comment]
@cuihtlauac I have removed the file you suggested and i have run the dune build --root . i am not getting any errors, please check again if its all good

File: practice/003/answer/impl.ml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# let rec at k = function
```
Comment (cuihtlauac) [inline practice/003/answer/impl.ml]:
```suggestion
# let rec at k = function
```
There is a syntax error here

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
Compilation still fails. 
```
>dune test --root . answer
File "answer/impl.ml", line 1, characters 2-5:
1 | # let rec at k = function
      ^^^
Error: Syntax error
```

**Comment by mnaibei** [comment]
@cuihtlauac how about now

**Review by cuihtlauac — CHANGES_REQUESTED** [review]
It does not compile `dune test --root . answer` fails

File: practice/003/ex.ml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    | [] -> None
+    | h :: _ -> Some h
+end
+
+module Answer : Testable = struct
+  let last lst = match List.rev lst with
+    | [] -> None
+    | h :: _ -> Some h
+end
+
```
Comment (cuihtlauac) [inline practice/003/ex.ml:25]:
```suggestion
module Answer : Testable = struct
  let last lst = match List.rev lst with
    | [] -> None
    | h :: _ -> Some h
end
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — APPROVED** [review]

**Comment by cuihtlauac** [comment]
Thanks @mnaibei

**Comment by mnaibei** [comment]
This has been quite the rollercoaster😂😂 @cuihtlauac, glad it is done now. I have one question by the way, In the final applications there are a few community-specific questions that I would appreciate your assistance on:

1. Some communities or projects may want you to answer additional questions. Please check with your mentor and community coordinator to see if you need to provide any additional information after you save your final application.

2. Project timeline. Please work with your mentor to provide a timeline of the work you plan to accomplish on the project and what tasks you will finish at each step.

how should I go about answering these?

**Pull request merged**